pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -941,11 +937,11 @@ func (vs *VSphere) HasClusterID() bool {
 // Notification handler when node is registered.
 func (vs *VSphere) NodeRegistered(node *v1.Node) {
 	glog.V(4).Infof("Node Registered: %+v", node)
-	vs.nodeManager.registerNode(node)
+	vs.nodeManager.RegisterNode(node)
```

Comment:
What happens if Registernode() is unable to register node?

Reply:
We put the node into a register list. So, from node manager's perspective we are aware of the node.

Reply:
So, nm.SetNode(node) will always pass. If nm.DiscoverNode(node) fails, it would not be a problem right?
After that for any node operations, you query GetNode(). It would throw error as it is not present in registered nodes. In case of error you again call nm.RediscoverNode(node), which we try to rediscover the node again. 

When RediscoverNode() fails you would throw an error in VCP. 
Correct me if I am wrong?

Reply:
Yes.

Reply:
nm.addNode will always pass. So, we know which nodes are registered. So nm.getNode will always work as long as we have received a notification of that node. Note that nm.GetNodeInfo may not work if we were not able to discover the node.

If GetNodeInfo does not find the nodeInfo for the node, then it makes sense to rediscover it. This is currently not done. @abrarshivani can you take care of it.

Reply:
@SandeepPissay will do that.